pkg/edgecache/cachevolume/pvc_annotator.go

```diff
@@ -122,7 +124,27 @@ func (c *PVCAnnotator) buildAnnotations(pv *v1.PersistentVolume, cfg config.Azur
 	return annotations, nil
 }
 
+func (c *PVCAnnotator) needsToBeProvisioned(pvc *v1.PersistentVolumeClaim) bool {
+	// check if pv connected to the pvc has already been passed to be created
+	pvState, pvStateOk := pvc.ObjectMeta.Annotations[createVolumeAnnotation]
+	if pvStateOk && pvState == "no" {
+		return false
+	}
+
+	return true
+}
+
 func (c *PVCAnnotator) SendProvisionVolume(pv *v1.PersistentVolume, cloudConfig config.AzureAuthConfig, providedAuth BlobAuth) error {
+	pvc, err := blobcsiutil.GetPVCByName(c.client, pv.Spec.ClaimRef.Name, pv.Spec.ClaimRef.Namespace)
+	if err != nil {
+		return err
+	}
+
+	if prepare := c.needsToBeProvisioned(pvc); !prepare {
```

Comment:
Nit: Since this is a single return value you can just do `if !c.needsToBeProvisioned(pvc)` right?

Reply:
changed!

https://github.com/avoltz/blob-csi-driver/blob/1135f3ab14b02930feb4bdc46940d2dd604a8159/pkg/edgecache/cachevolume/pvc_annotator.go#L143-L144